src/args.rs

```diff
@@ -49,17 +64,48 @@ impl Args {
         self.paths.is_empty()
     }
 
-    pub(crate) fn output_mode(&self) -> OutputMode {
-        if self.vimgrep {
-            OutputMode::Vimgrep
+    fn line_number(&self) -> bool {
+        true
+    }
+
+    fn per_match(&self) -> bool {
```

Comment:
This would basically correspond to a "knob" for "should we print every match (vs congeal adjacent/overlapping match lines)", which seems like as we've discussed would maybe be useful(/expected? Should it be the default?) for our use case